addons/website_sale/models/website.py

```diff
@@ -568,74 +568,63 @@ def _send_abandoned_cart_email(self):
                 template.send_mail(sale_order.id, email_values={'email_to': sale_order.partner_id.email})
                 sale_order.cart_recovery_email_sent = True
 
-    def _get_checkout_step_list(self):
-        """ Return an ordered list of steps according to the current template rendered.
+    @api.model_create_multi
+    def create(self, vals_list):
+        websites = super().create(vals_list)
+        for website in websites:
+            website._create_checkout_steps()
+        return websites
+
+    def _create_checkout_steps(self):
+        generic_steps = self.env['website.checkout.step'].sudo().search([
+            ('website_id', '=', False),
+        ])
+        for step in generic_steps:
+            is_published = bool(step.step_href != '/shop/extra_info')
+            step.copy({'website_id': self.id, 'is_published': is_published})
+
+    def _get_website_checkout_step(self, href):
+        return self.env['website.checkout.step'].sudo().search([
+            ('website_id', '=', self.id),
+            ('step_href', '=', href),
+        ], limit=1)
```

Comment:
Why not simply `_get_checkout_step`?
```suggestion
    def _get_checkout_step(self, href):
        return self.env['website.checkout.step'].sudo().search([
            ('website_id', '=', self.id),
            ('step_href', '=', href),
        ], limit=1)
```